### DISCUSSION TITLE
Deployed NextJS app's API returns 404

### DISCUSSION THREAD

**Discussion by nikolagigic** [vercel/next.js#41781] [Help] [answered]
Hey everyone,
So I've been struggling with this for past day and can't figure out why when I try to make a POST request to my API endpoint it keeps on returning 404 when the application is deployed on Vercel - it works perfectly fine locally.
I went through all [checks](https://vercel.com/guides/why-does-my-serverless-function-work-locally-but-not-when-deployed) and sorted them out if there were any on why my Serverless Function might not work and I still get the same issue.

Any idea why else could this happen?

Front-end is sending a `Blob` through `FormData` to back-end which then uses `formidable` to parse the file and perform side-effects with it. App is in **Turborepo**.

**Comment by chanmyaemaung** [comment]
Same to you. Did you solve? I am also stacked as the same like you'd facing this issues on the productions. Even is a just simple api routes but that returned as 404 not found in the production.

**Comment by kamenskiyyyy** [comment]
Maybe this is a problem in version 13? I also have this problem.

**Reply by saad-devx** [reply]
Hi, i am having the same issue, Pleassee help me if you figured it out somehow.

**Comment by nikolagigic** [comment, marked as answer]
The issue was not with API routes - the issue is that Vercel is masking unhandled exceptions to `404`. So if something bad happens on back-end and you do not catch the exception **AND** returrn it to front-end you will never know what actually happened.

**Comment by AyushUnleashed** [comment]
Okay, so to fix this issue:
1. Handle exceptions better in the nextjs backend - your nextjs backend is doing some operation & failing at it on prod but not on local, it could be due to env issue. ( It was in my case )
2. once you fix the operation on your backend, it would work, it would give back response & 404 would go away.

**Comment by huisnotacouncillor** [comment]
Facing the same issue, it works most of the time, but sometimes it will return 404 page as the request's response,  annoying.
